core/src/test/scala/unit/kafka/admin/DeleteTopicTest.scala

```diff
@@ -187,264 +220,337 @@ class DeleteTopicTest extends QuorumTestHarness {
-    servers = createTestTopicAndCluster(topic)
+    createTestTopicAndCluster(topic)
     // start topic deletion
-    adminZkClient.deleteTopic(topic)
-    TestUtils.verifyTopicDeletion(zkClient, topic, 1, servers)
+    adminClient.deleteTopics(Collections.singleton(topic))
+    TestUtils.verifyTopicDeletion(zkClientOrNull, topic, 1, brokers)
     // re-create topic on same replicas
-    TestUtils.createTopic(zkClient, topic, expectedReplicaAssignment, servers)
+    TestUtils.createTopicWithAdmin(admin = adminClient, topic = topic, brokers = brokers, replicaAssignment = expectedReplicaAssignment)
     // check if all replica logs are created
-    TestUtils.waitUntilTrue(() => servers.forall(_.getLogManager.getLog(topicPartition).isDefined),
+    TestUtils.waitUntilTrue(() => brokers.forall(_.logManager.getLog(topicPartition).isDefined),
       "Replicas for topic test not created.")
   }
 
-  @Test
-  def testDeleteNonExistingTopic(): Unit = {
+  @ParameterizedTest(name = TestInfoUtils.TestWithParameterizedQuorumName)
+  @ValueSource(strings = Array("zk", "kraft"))
+  def testDeleteNonExistingTopic(quorum: String): Unit = {
     val topicPartition = new TopicPartition("test", 0)
     val topic = topicPartition.topic
-    servers = createTestTopicAndCluster(topic)
+    createTestTopicAndCluster(topic)
     // start topic deletion
-    assertThrows(classOf[UnknownTopicOrPartitionException], () => adminZkClient.deleteTopic("test2"))
-    // verify delete topic path for test2 is removed from ZooKeeper
-    TestUtils.verifyTopicDeletion(zkClient, "test2", 1, servers)
+    adminClient.deleteTopics(Collections.singleton("test2"))
+    if (!isKRaftTest()) {
+      // verify delete topic path for test2 is removed from ZooKeeper
+      TestUtils.verifyTopicDeletion(zkClientOrNull, "test2", 1, brokers)
+    }
     // verify that topic test is untouched
-    TestUtils.waitUntilTrue(() => servers.forall(_.getLogManager.getLog(topicPartition).isDefined),
+    TestUtils.waitUntilTrue(() => brokers.forall(_.logManager.getLog(topicPartition).isDefined),
       "Replicas for topic test not created")
-    // test the topic path exists
-    assertTrue(zkClient.topicExists(topic), "Topic test mistakenly deleted")
-    // topic test should have a leader
-    TestUtils.waitUntilLeaderIsElectedOrChanged(zkClient, topic, 0, 1000)
+    if (isKRaftTest()) {
+      // test the topic exists
+      val partition = getTopicPartitionInfo(adminClient, topic, 0)
+      assertTrue(partition.isPresent, "Partition [test,0] should exist")
+      // topic test should have a leader
+      assertFalse(partition.get().leader().isEmpty, "Leader should exist for partition [test,0]")
+    } else {
+      // test the topic path exists
+      assertTrue(zkClient.topicExists(topic), "Topic test mistakenly deleted")
+      // topic test should have a leader
+      TestUtils.waitUntilLeaderIsElectedOrChanged(zkClient, topic, 0, 1000)
+    }
```

Comment:
Please avoid this pattern. Can't you just use the Admin client for both changes? If not please document why.

Reply:
For the most part I tried to honor the original checks from the original ZK tests (e.g. checks that the topic path in ZK exist), I guess in many cases other checks are sufficient.